src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs

```diff
@@ -134,31 +128,17 @@ public class AdminController : Controller
                 return View(model);
             }
 
-            var app = await _applicationManager.FindByIdAsync(model.Id);
-            if (app == null)
+            var application = await _applicationManager.FindByIdAsync(model.Id, HttpContext.RequestAborted);
+            if (application == null)
                 return NotFound();
 
-            app.DisplayName = model.DisplayName;
-            app.RedirectUri = model.RedirectUri;
-            app.LogoutRedirectUri = model.LogoutRedirectUri;
-            app.ClientId = model.ClientId;
-            app.Type = model.Type;
-            app.SkipConsent = model.SkipConsent;
-            app.RoleNames = new List<string>();
-            if (app.Type == ClientType.Confidential)
-                app.RoleNames = model.RoleEntries.Where(r => r.Selected).Select(r => r.Name).ToList();
-            app.AllowAuthorizationCodeFlow = model.AllowAuthorizationCodeFlow;
-            app.AllowClientCredentialsFlow = model.AllowClientCredentialsFlow;
-            app.AllowImplicitFlow = model.AllowImplicitFlow;
-            app.AllowPasswordFlow = model.AllowPasswordFlow;
-            app.AllowRefreshTokenFlow = model.AllowRefreshTokenFlow;
-            app.AllowHybridFlow = model.AllowHybridFlow;
-
-            await _applicationManager.CreateAsync(app);
+            await TryUpdateModelAsync(application);
+            await _applicationManager.UpdateAsync(application, HttpContext.RequestAborted);
```

Comment:
Yeah. We'll likely need to update OpenIddict to expose an `UpdateAsync` overload accepting the client secret parameter.